### PR TITLE
Deployment improvement

### PR DESCRIPTION
**Pull request by josojo**
Some modifications to the deployment scripts for easier runnablity and readablity.

(Unfortuantley, verifiying the bridge contract is still not possible.)

### CODE REVIEW

File: src/deployment/1_deployforkableToken.js [hunk 1]
```diff
@@ -16,33 +16,39 @@ async function main() {
...
                 }
-                currentProvider.getFeeData = overrideFeeData;
             }
         }
     }
     // Load deployer
     let deployer;
     if (deployParameters.deployerPvtKey) {
         deployer = new ethers.Wallet(deployParameters.deployerPvtKey, currentProvider);
+    } else if (process.env.HARDHAT_NETWORK === 'localhost') {
```
Comment (josojo) [inline src/deployment/1_deployforkableToken.js:50]: This option makes it easier to deploy to the localhost...

File: src/deployment/1_deployforkableToken.js [hunk 2]
```diff
@@ -16,33 +16,39 @@ async function main() {
...
-                };
-                currentProvider.getFeeData = async () => FEE_DATA;
-            } else {
-                console.log('Multiplier gas used: ', deployParameters.multiplierGas);
-                async function overrideFeeData() {
-                    const feedata = await ethers.provider.getFeeData();
-                    return {
-                        maxFeePerGas: feedata.maxFeePerGas.mul(deployParameters.multiplierGas).div(1000),
-                        maxPriorityFeePerGas: feedata.maxPriorityFeePerGas.mul(deployParameters.multiplierGas).div(1000),
+    if (process.env.HARDHAT_NETWORK === 'localhost') {
```
Comment (josojo) [inline src/deployment/1_deployforkableToken.js:19]: Adding config for localhost... not sure why it worked before without this helper

File: src/deployment/2_creategenesis.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ async function main() {
      * deploy proxy
      * Do not initialize directlythe proxy since we want to deploy the same code on L2 and this will alter the bytecode deployed of the proxy
      */
-    const transparentProxyFactory = await ethers.getContractFactory('TransparentUpgradeableProxy', deployer);
+    const transparentProxyFactory = await ethers.getContractFactory('@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol:TransparentUpgradeableProxy', deployer);
```
Comment (josojo) [inline src/deployment/2_creategenesis.js:126]: One of the side effects of having two compiled contracts: (one from foundry and one from hardhat) is that we need to specifiy it more in detail

File: src/deployment/3_deployContracts.js [hunk 4]
```diff
@@ -403,36 +410,13 @@ async function main() {
...
-            ]);
-            const dataCallInitialize = iface.encodeFunctionData(
-                'initialize(address,address,address,address)',
-                [
-                    forkingManagerContract.address,
-                    parentContract,
-                    precalculateZkevmAddress,
-                    proxyBridgeAddress,
-                ],
+            const iForkableExitRoot = await ethers.getContractAt('IForkableGlobalExitRoot', polygonZkEVMGlobalExitRoot.address);
```
Comment (josojo) [inline src/deployment/3_deployContracts.js:413]: This is a much more consice formulation that is also easier to refactor, than writing the abi function down into code, as we did before

**Review by edmundedgar — APPROVED** [review]

**Pull request merged**